pkgs/build-support/ocaml/dune.nix

```diff
@@ -34,12 +34,18 @@ stdenv.mkDerivation ({
     runHook postInstall
   '';
 
-} // (builtins.removeAttrs args [ "minimalOCamlVersion" "duneVersion" ]) // {
+  strictDeps = true;
```

Comment:
given the feature freeze https://github.com/NixOS/nixpkgs/issues/167025 I think this should be separated to a PR after branch off.

Reply:
Actually, I would be more concerned about the first commit that adds `checkInputs` to `buildInputs` if `doCheck` is true :thinking:
Since this PR already makes sure all the current packages work with `strictDeps = true` it should be fine to merge it this way?

Reply:
Oh I had not seen that... I'm not a fan of this kind of workaround honestly.